### ISSUE TITLE
[Go] How to use multiple Providers?

### ISSUE THREAD

**Issue by elsesiy** [pulumi/pulumi-azure#461]
Hi,

consider the following scenario:
I have two vnets residing under the same tenant in different subscriptions. 
My service principal has the necessary role assignments to access both resources.
In terraform I have to create two separate providers in order to establish vnet peering successfully.

How can I achieve the same mechanism in Pulumi (Go)?

Thanks.

**Comment by mikhailshilkov** [comment]
@EvanBoyle Have we added support for explicit providers to the Go SDK yet?

**Comment by EvanBoyle** [comment]
@elsesiy You can specify an explicit provider. 

```go
var provider ... // instantiate an explicit azure provider
pulumi.Provider(provider) // returns a resourceOption to pass into your NewResource call.
```

This was added recently and [the docs](https://www.pulumi.com/docs/intro/concepts/programming-model/#explicit-provider-configuration
) haven't been updated yet. I'll get that fixed ASAP

**Comment by elsesiy** [comment]
@EvanBoyle Great, thank you. I'll give this is a shot and report back!

**Comment by elsesiy** [comment]
Unfortunately I'm running into some issues here. This is the provider definition:
```
		peerProvider, err := azure.NewProvider(ctx, "peering-provider", &azure.ProviderArgs{
			SubscriptionId:            pulumi.String(resourceConfig.Get("peer_subscription_id")),
		})
```

Diagnostics:
```
  pulumi:providers:azure (peering-provider):
    error: pulumi:providers:azure resource 'peering-provider's property 'skipCredentialsValidation' value {false} has a problem: provider property values must be strings
    error: pulumi:providers:azure resource 'peering-provider's property 'useMsi' value {false} has a problem: provider property values must be strings
    error: pulumi:providers:azure resource 'peering-provider's property 'disableTerraformPartnerId' value {true} has a problem: provider property values must be strings
    error: pulumi:providers:azure resource 'peering-provider's property 'skipProviderRegistration' value {false} has a problem: provider property values must be strings
```

I also tried explicitly specifying these properties but since they are all of type `pulumi.BoolPtrInput` the result is the same.

**Comment by EvanBoyle** [comment]
This is a known issue with the code generation for providers. We've got a branch with a fix we're working on: https://github.com/pulumi/pulumi/compare/pgavlin/providerConfig

**Comment by elsesiy** [comment]
Okay is that fix also going into `1.11` or what's the ETA for it? Thanks for the quick support, much appreciated!

**Comment by EvanBoyle** [comment]
We have a PR out https://github.com/pulumi/pulumi/pull/3948 but the 1.11 release is already baking so this won't make it in. 

In the meantime, if you'd like to unblock yourself you can build this branch locally by following [contributing.md](https://github.com/pulumi/pulumi/blob/master/CONTRIBUTING.md)

**Comment by stack72** [comment]
Hi @elsesiy 

This will have been released in 1.11.1 - I am going to close this out as @EvanBoyle has fixed this up

Please let me know if there are any issues

Paul
